api/app/Http/Controllers/UserController.php

```diff
@@ -18,17 +56,335 @@
+     * @throws Exception
+     */
+    public function index(Request $request): JsonResponse
+    {
+        /** @var UserLaravel $currentUser */
+        $currentUser = auth()->user();
+        
+        if ($currentUser->roleId !== $this->roles->Admin) {
+            return response()->json(new ErrorResponse(Response::$statusTexts[403]), 403);
+        }
+        
+        return response()->json(new UserCollection(UserLaravel::all()), 200);
+    }
+    
+    /**
+     * Post /users
+     *
+     * Create new User
+     *
+     * @param Request $request
+     *
+     * @return JsonResponse
+     */
+    public function store(Request $request): JsonResponse
+    {
+        /** @var UserLaravel $currentUser */
+        $currentUser = auth()->user();
+        
+        if ($currentUser->roleId !== $this->roles->Admin) {
+            return response()->json(new ErrorResponse(Response::$statusTexts[403]), 403);
+        }
+        
+        $count = $request->query('count') ? (int)$request->query('count') : 1;
+        if ($count > 500) {
+            return response()->json(new ErrorResponse('Count is to high'), 400);
+        }
+        
+        $users = new Collection();
+        for ($i = 0; $i < $count; $i++) {
+            /** @var UserLaravel $user */
+            $user = UserLaravel::query()->make();
+            $user->roleId = $this->roles->Customer;
+            $user->apiKey = md5(Guid::uuid4()->toString());
+            $user->isBlocked = false;
+            
+            $user->save();
+            
+            $users->add($user);
+        }
+        
+        $userLinks = $users->map(function (UserLaravel $user) {
+            return "https://lk.agishev-autoz.ru/tables?hash=".$user->apiKey;
+        })->implode(PHP_EOL);
+        mail('maksimagishev@mail.ru', 'Новые пользователи', $userLinks);
+        
+        return response()->json(new UserCollection($users), 201);
+    }
+    
+    /**
+     * Put /users/{$id}
+     *
+     * @param Request $request
+     * @param         $id
+     *
+     * @return JsonResponse
+     * @throws Exception
+     */
+    public function update(Request $request, $id): JsonResponse
+    {
+        /** @var UserLaravel $currentUser */
+        $currentUser = auth()->user();
+    
+        Log::channel('apilog')->info('PUT /users/'.$id.' - user '.$currentUser->id);
+        
+        if (!(($currentUser->id === (int)$id) ||
+            ($currentUser->roleId === $this->roles->Admin))) {
+            return response()->json(new ErrorResponse(Response::$statusTexts[403]), 403);
+        }
+        
+        /** @var UserLaravel|null $user */
+        $user = UserLaravel::query()->find((int)$id);
+        if (is_null($user)) {
+            return response()->json(new ErrorResponse(Response::$statusTexts[404]), 404);
+        }
+        
+        $request->validate([
+            'phoneNumber' => 'string|nullable',
+            'socialNetworkUrl' => 'string|nullable',
```

Comment:
These values require a bit more of validation or some phone numbers will be 'Unicorn666' or something along these lines